docs/MongoDB-batchsource.md

```diff
@@ -4,30 +4,41 @@
 Description
 -----------
 Reads documents from a MongoDB collection and converts each document into a StructuredRecord with the help
-of a specified schema. The user can optionally provide input query, input fields, and splitter classes.
+of a specified schema. The user can optionally provide input query.
 
 
 Configuration
 -------------
-**referenceName:** This will be used to uniquely identify this source for lineage, annotating metadata, etc.
+**Reference Name:** Name used to uniquely identify this source for lineage, annotating metadata, etc.
 
-**connectionString:** MongoDB connection string. Example: `mongodb://localhost:27017/analytics.users` (Macro-enabled)
-[Reference](http://docs.mongodb.org/manual/reference/connection-string)
+**Host:** Host that MongoDB is running on.
 
-**schema:** Specifies the schema of the documents.
+**Port:** Port that MongoDB is listening to.
 
-**authConnectionString:** Auxiliary MongoDB connection string to authenticate against when constructing splits. (Macro-enabled)
+**Database:** MongoDB database name.
 
-**inputQuery:** Optionally filter the input collection with a query. This query must be represented in JSON format
-and use the MongoDB extended-JSON format to represent non-native JSON data types. (Macro-enabled)
+**Collection:** Name of the database collection to read from.
 
-**inputFields:** Projection document that can limit the fields that appear in each document. 
-If no projection document is provided, all fields will be read. (Macro-enabled)
+**Output Schema:** Specifies the schema of the documents.
 
-**splitterClass:** The name of the Splitter class to use. If left empty, the MongoDB Hadoop Connector will attempt
-to make a best-guess as to which Splitter to use. (Macro-enabled) The Hadoop connector provides these Splitters:
+**Input Query:** Optionally filter the input collection with a query. This query must be represented in JSON format
+and use the [MongoDB extended JSON format] to represent non-native JSON data types. (Macro-enabled)
 
-  - `com.mongodb.hadoop.splitter.StandaloneMongoSplitter`
-  - `com.mongodb.hadoop.splitter.ShardMongoSplitter`
-  - `com.mongodb.hadoop.splitter.ShardChunkMongoSplitter`
-  - `com.mongodb.hadoop.splitter.MultiMongoCollectionSplitter`
+**Username:** User identity for connecting to the specified database.
+
+**Password:** Password to use to connect to the specified database.
+
+**On Record Error:** Specifies how to handle error in record processing. An error will be thrown if failed to parse
+value according to a provided schema.
+
+**Authentication Connection String:** Auxiliary MongoDB connection string to authenticate against when constructing
+splits. (Macro-enabled)
```

Comment:
The [documentation says](https://github.com/mongodb/mongo-hadoop/wiki/Configuration-Reference#mongoauthuri-specify-an-auxiliary) that this Connection String should be used in the case of a sharded cluster with `config` database requiring authentication(if its username/password differs from the ones used to access the collection).

This URI is also used by `com.mongodb.hadoop.splitter.MongoSplitterFactory` to determine what Splitter implementation to use.

Description changed to:
```
**Authentication Connection String:** Optional MongoDB connection string to connect to the 'config' database of a
sharded cluster. It can be omitted if username and password do not differ from the previously provided ones or if
'config' database does not require authentication.
```